inception-versioning/src/main/java/de/tudarmstadt/ukp/inception/versioning/ui/VersioningSettingsPanel.java

```diff
@@ -0,0 +1,217 @@
+            LOG.error("Error getting remote: {}", e.getMessage());
+            error("Error getting remote: " + ExceptionUtils.getRootCauseMessage(e));
+            return;
+        }
+
+        repositoryConfigModel = new CompoundPropertyModel<>(repositoryConfig);
+
+        // Settings settingsForm
+        Form<RepositoryConfig> settingsForm = new Form<>("settingsForm", repositoryConfigModel);
+        settingsForm.setOutputMarkupId(true);
+        add(settingsForm);
+
+        // Local path
+        TextField<String> localPathField = new TextField<>("localPath");
+        settingsForm.add(localPathField);
+
+        // Remote path
+        settingsForm.add(new TextField<String>("remotePath").add(new UrlValidator()));
+        settingsForm.add(new LambdaAjaxButton<>("setRemoteButton", this::actionSetRemote));
+
+        // Buttons
+        settingsForm.add(new LambdaAjaxLink("snapshotProject", this::actionSnapshotProject));
+
+        // Push form
+        PushConfig pushConfig = new PushConfig();
+        pushConfigModel = new CompoundPropertyModel<>(pushConfig);
+        Form<PushConfig> pushForm = new Form<>("pushForm", pushConfigModel);
+        pushForm.add(new TextField<>("username"));
+        pushForm.add(new PasswordTextField("password"));
+        pushForm.add(new LambdaAjaxButton<>("pushButton", this::actionPushToRemote));
+        add(pushForm);
+    }
+
+    private void actionSetRemote(AjaxRequestTarget aTarget, Form<RepositoryConfig> aForm)
+    {
+        try {
+            versioningService.setRemote(getModelObject(),
+                    repositoryConfigModel.getObject().remotePath);
+            aTarget.add(this);
+            info("Setting remote successful!");
+        }
+        catch (IOException | GitAPIException | URISyntaxException e) {
+            LOG.error("Error setting remote: {}", e.getMessage());
+            error("Error setting remote: " + ExceptionUtils.getRootCauseMessage(e));
+        }
+        aTarget.addChildren(getPage(), IFeedback.class);
+    }
+
+    private void actionSnapshotProject(AjaxRequestTarget aTarget)
+    {
+        try {
+            versioningService.snapshotCompleteProject(getModelObject(), "Snapshotting");
+            aTarget.add(this);
+            info("Snapshotting successful!");
+        }
+        catch (IOException | GitAPIException e) {
+            LOG.error("Error snapshotting project: {}", e.getMessage());
+            error("Error snapshotting project: " + ExceptionUtils.getRootCauseMessage(e));
+        }
+
+        aTarget.addChildren(getPage(), IFeedback.class);
+    }
+
+    private void actionPushToRemote(AjaxRequestTarget aTarget, Form<RepositoryConfig> aForm)
+    {
+        try {
+            PushConfig pushConfig = pushConfigModel.getObject();
+            versioningService.pushToOrigin(getModelObject(), pushConfig.getUsername(),
+                    pushConfig.getPassword());
+            aTarget.add(this);
+            info("Pushing successful!");
```

Comment:
There is also a `success(msg)` call which would seem more appropriate to report a success than the `info(msg)` call.

Reply:
Done